### ISSUE TITLE
draw_texture_rotate changes x,y of texture

### ISSUE THREAD

**Issue by georgesofianosgr** [xerpi/sf2dlib#14]
"draw_texture_rotate" uses as the center of rotation x & y of the texture
and moves the texture half of its width left and half of its height up.

![capture](https://cloud.githubusercontent.com/assets/2379934/9832308/43a5bcd6-597f-11e5-8952-8b950e816735.PNG)
![capture2](https://cloud.githubusercontent.com/assets/2379934/9832310/446c4270-597f-11e5-98f1-1561d9601e5e.PNG)

**Comment by xerpi** [comment]
Yes I know, this is the expected behaviour since draw_texture_rotate is simply calling draw_texture_rotate_hotspot.

**Comment by georgesofianosgr** [comment]
Thanks for the quick response. So this is not a bug ?
The description of the function is "Draws a texture with rotation around its center"
When I enter x & y of the texture I set the top left corner of the texture. 
So the center of the texture shouldn't be "x + width / 2.0" and "y + height / 2.0"

**Comment by xerpi** [comment]
I'm a bit confused, draw_texture_rotate will rotate around the center of the texture, and this center will be drawn at  (x, y).

**Comment by georgesofianosgr** [comment]
Ok I get it  now, thank you.

It's a little bit confusing that all the other draw_texture functions that don't use rotation, their x & y parameters are for setting the upper left corner of the texture but in rotation x & y parameters are the center of rotation. I expected that x & y parameters are still used for the upper left corner and the texture's rotation center would be "x + texture->width / 2.0" and "y + texture->height / 2.0"

**Comment by xerpi** [comment]
Yeah I understand it, maybe I should change it but I don't know if there are any other projects depending on that.

**Comment by georgesofianosgr** [comment]
I think it's expected to have changes like that in an under development library. 
Personally I always check the commits before updating the library. 

Maybe you need to do some "stable" releases so most people will stick with that. 
Some small releases for critical things like a libctru update so when people update 
to that will not have anything broken in their projects
and the master branch will have all the heavy changes.

Also if you decide not to change the code for the rotation I would suggest you to update the function description so people are aware that x & y is the center of the texture.
